plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/properties/PropertySourceCustom.java

```diff
@@ -98,6 +99,8 @@ public void addDefaultValues(Map<String, ?> defaultValues)
     public Map<String, Object> getPropertyValues() {
         Map<String, Object> allValues = new HashMap<>(originalValues);
         allValues.putAll(propValues);
+        allValues.values().forEach(it -> GeneralUtils.replaceVariables(CommonUtils.toString(it),
+            SystemVariablesResolver.INSTANCE));
```

Comment:
I'm suspicious about converting property values from * something * into string just to replace variables.

If variable values can only be strings, then it's better to return `Map<String, String>` instead.